Lombiq.HelpfulLibraries/Libraries/Utilities/StringExtensions.cs

```diff
@@ -27,18 +27,20 @@ public static class StringExtensions
         /// A shortcut for <c>string.Contains(string, StringComparison.OrdinalIgnoreCase)</c>.
         /// </summary>
         public static bool ContainsOrdinalIgnoreCase(this string? text, string? value) =>
-            text != null && !string.IsNullOrEmpty(value) && text.Contains(value, StringComparison.OrdinalIgnoreCase);
+            text != null && value != null && text.Contains(value, StringComparison.OrdinalIgnoreCase);
 
         /// <summary>
         /// A shortcut for <c>string.StartsWith(string, StringComparison.OrdinalIgnoreCase)</c>.
         /// </summary>
         public static bool StartsWithOrdinalIgnoreCase(this string? text, string? value) =>
-            text != null && !string.IsNullOrEmpty(value) && text.StartsWith(value, StringComparison.OrdinalIgnoreCase);
+            text != null && value != null && text.StartsWith(value, StringComparison.OrdinalIgnoreCase);
 
         /// <summary>
-        /// A shortcut for <c>string.Equals(string, StringComparison.OrdinalIgnoreCase)</c>.
+        /// A shortcut for <c>string.Equals(string, StringComparison.OrdinalIgnoreCase)</c>. Also returns <see
+        /// langword="true"/> if both are <see langword="null"/>.
         /// </summary>
         public static bool EqualsOrdinalIgnoreCase(this string? text, string? value) =>
-            text != null && !string.IsNullOrEmpty(value) && text.Equals(value, StringComparison.OrdinalIgnoreCase);
+            (text == null && value == null) ||
+            (text != null && value != null && text.Equals(value, StringComparison.OrdinalIgnoreCase));

```

Comment:
On a second thought, these shortcuts should just be shortcuts indeed. No further checks, just proxy the call; if there's an exception from the original method then there should be. No?

Reply:
I think it's more sensible to be symmetrical. That is, if value can be null then why can't text be? Particularly here, since "null equals null?" is a plausible scenario and true is a logical result. The stock method throws an NRE for technical reasons (contrast with operator overloading) rather than because it's the logical thing to do.

I think if we omit the null checks then there is not much value in just having a shortcut extension method that's almost as long as the line it replaces. This doesn't strike me as a huge save:
```csharp
text.EqualsOrdinalIgnoreCase(value);
text.Equals(value, StringComparison.OrdinalIgnoreCase);
```

Reply:
It's a small saving but it's still one, especially when you type it ("text.EqualsO tab tab" vs parameterizing the method call). These shortcuts needn't aim for some huge goals :). I think the API is misleading if it promises to be a shortcut but then it tries to be a "better" version; in the latter case I'd name these differently too (but then again I think they should be just simple shortcuts).

Reply:
Okay then.